config/prow/config.yaml

```diff
@@ -466,6 +466,12 @@ branch-protection:
             - stage-bots
         website:
           protect: true
+          enforce_admins: false
+          required_status_checks:
+            contexts:
+              - EasyCLA
+          exclude:
+            - "^dev-" # don't protect dev branches
```

Comment:
The docs release team handbook used to mention explicitly to protect the `dev-` branch: https://github.com/kubernetes/sig-release/blob/e57c4d957e371dd9a96dc7ba0ae8b5cadd5928bd/release-team/role-handbooks/docs/Release-Timeline.md#enable-branch-protection . This was only recently removed by me assuming that all branches are protected by default, however the new docs role handbook still says the branch should be protected (https://github.com/kubernetes/sig-release/blob/master/release-team/role-handbooks/docs/Release-Timeline.md#modify-prow-config-file):

> e.g. Create a https://github.com/kubernetes/test-infra/pull/21727 against [k/test-infra](https://github.com/kubernetes/test-infra) to configure [prow](https://github.com/kubernetes/test-infra/tree/master/prow#) to automatically apply milestones to future release branch and to remove the configuration for the last release e.g. Add branch protection and milestone applier for k/website.

This might become misleading after this change.

Reply:
Prow experts,
Can we explicitly add `dev-` branches to have branch protection?

Reply:
By default, all branches would be protected. Is there something special about this branch that it shouldn't have that protection?

Reply:
I want to be able to delete a localization branch - see https://github.com/kubernetes/test-infra/pull/30394#issuecomment-1680927286

I don't know how we delete stale localization branches nor how we delete the dev branches for previous releases. Is there a process that can work without making this change?

Reply:
Another example for deleting specific branches in k/website
At the end of the 1.27 release cycle, a `release-1.27` branch was incorrectly made so I deleted it
The same thing happened last week for the 1.28 release cycle, a `release-1.28` branch was incorrectly made and now we have to delete it before the end of the 1.29 release cycle (which is months away at this point)